codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -50,29 +50,31 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         this.connection = checkNotNull(connection, "connection");
 
         // Add a flow state for the connection.
-        connection.connectionStream().setProperty(FlowState.class,
-                new FlowState(connection.connectionStream(), initialWindowSize));
+        connection.connectionStream().remoteFlowState(
+                new DefaultFlowState(new ReducedFlowState(connection.connectionStream()), initialWindowSize));
 
         // Register for notification of new streams.
         connection.addListener(new Http2ConnectionAdapter() {
-            @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.setProperty(FlowState.class, new FlowState(stream, 0));
+                // TODO: (scottmitch): we used to have a create and open...but now they are separate.
```

Comment:
This may be problematic and create GC pressure.  I'm not sure if we need to do something "fancy" here to directly allocate `DefaultFlowState` if stream is created and opened at the same time.